libraries/AP_Common/Location.cpp

```diff
@@ -355,8 +355,7 @@ bool Location::sanitize(const Location &defaultLoc)
 
     // convert relative alt=0 to mean current alt
     if (alt == 0 && relative_alt) {
-        relative_alt = false;
-        alt = defaultLoc.alt;
+        set_alt_cm(defaultLoc.alt, defaultLoc.get_alt_frame());
```

Comment:
This is an intriguing change - looks good to my eye - but, oh boy, does it have implications!  I'm tempted to suggest you pull this out as a separate PR we can discuss at a DevCall because this is going to make people nervous.

Reply:
Created 20122